lib/logstash/filters/geoip.rb

```diff
@@ -124,22 +124,24 @@ def register
 
   public
   def filter(event)
-    
+
     geo_data = nil
 
     geo_data = get_geo_data(event)
 
-    return if geo_data.nil? || !geo_data.respond_to?(:to_hash)
-
-    apply_geodata(geo_data, event)
+    return unless geo_data.respond_to?(:to_hash)
 
-    filter_matched(event)
+    event[@target] = {} if event[@target].nil?
+    geo_data_hash = geo_data.to_hash
+    # don't do anything more if the lookup result is empty
+    if !geo_data_hash.empty?
```

Comment:
True but as I decided that `apply_geodata` should expect and operate on a hash, I moved the hash generation to the `filter` method.  So I test the hash for empty.

One could easily argue that the `get_geo_data_for_ip` method should return a hash, meaning that we cache the hash.  Indeed the mutation of the hash in lines 144 - 148 should be done once before it is stored in the cache.  No need to do it each time, especially if there are many events referencing the same IP address.